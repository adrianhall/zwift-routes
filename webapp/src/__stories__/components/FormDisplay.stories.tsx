import React from 'react';
import { action } from '@storybook/addon-actions';
import FormDisplay from 'src/ui/components/FormDisplay';
import { DisplayUnits, DisplaySettings } from 'src/models';

export default {
  title: 'Components/FormDisplay'
};

const displaySettings: DisplaySettings = {
  units: DisplayUnits.Imperial
};

export const normal: React.SFC<{}> = () => (
  <div style={{ width: '320px', background: '#F0F0F0' }}>
    <FormDisplay onUpdate={action('onUpdate')} displaySettings={displaySettings} />
  </div>
);
