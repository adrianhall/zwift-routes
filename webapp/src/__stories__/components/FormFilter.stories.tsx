import React from 'react';
import { action } from '@storybook/addon-actions';
import FormFilter from 'src/ui/components/FormFilter';
import { RouteFilter, ZwiftRoute } from 'src/models';
import routeData from 'src/data/routes.json';

export default {
  title: 'Components/FormFilter'
};

const filter: RouteFilter = {
  includeCompletedRoutes: false,
  includeDefaultWorld: true,
  includeEventOnlyRoutes: false
};

const routes = routeData.map((r) => new ZwiftRoute(r));

export const normal: React.SFC<{}> = () => (
  <div style={{ width: '320px', background: '#F0F0F0' }}>
    <FormFilter onUpdate={action('onUpdate')} filter={filter} routes={routes} />
  </div>
);
