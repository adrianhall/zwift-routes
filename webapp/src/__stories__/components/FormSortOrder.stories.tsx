import React from 'react';
import { action } from '@storybook/addon-actions';
import FormSortOrder from 'src/ui/components/FormSortOrder';
import { SortDirection, SortField, SortOrder } from 'src/models';

export default {
  title: 'Components/FormSortOrder'
};

const sortOrder: SortOrder = {
  sortField: SortField.RouteDistance,
  sortDirection: SortDirection.Ascending
};

export const normal: React.SFC<{}> = () => (
  <div style={{ width: '320px', background: '#F0F0F0' }}>
    <FormSortOrder onUpdate={action('onUpdate')} sortOrder={sortOrder} />
  </div>
);
