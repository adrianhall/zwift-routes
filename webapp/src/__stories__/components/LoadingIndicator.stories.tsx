import React from 'react';
import { action } from '@storybook/addon-actions';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import LoadingIndicator from 'src/ui/components/LoadingIndicator';

export default {
  title: 'Components/LoadingIndicator'
};

// Sub-story - normal usage
export const normal: React.SFC<{}> = () => (
  <AppBar>
    <Toolbar>
      <Typography style={{ flexGrow: 1 }} variant="h6" noWrap>Normal</Typography>
      <LoadingIndicator />
    </Toolbar>
  </AppBar>
);

// Sub-story - is loading
export const loading: React.SFC<{}> = () => (
  <AppBar>
    <Toolbar>
      <Typography style={{ flexGrow: 1 }} variant="h6" noWrap>Loading</Typography>
      <LoadingIndicator isLoading={true} />
    </Toolbar>
  </AppBar>
);

// Sub-story - is loading
export const error: React.SFC<{}> = () => (
  <AppBar>
    <Toolbar>
      <Typography style={{ flexGrow: 1 }} variant="h6" noWrap>Error</Typography>
      <LoadingIndicator error={new Error('error message')} onClick={action('onClick')} />
    </Toolbar>
  </AppBar>
);
