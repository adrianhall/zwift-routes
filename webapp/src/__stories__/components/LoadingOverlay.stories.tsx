import React from 'react';
import LoadingOverlay from 'src/ui/components/LoadingOverlay';

export default {
  title: 'Components/LoadingOverlay'
};

// Sub-story - normal usage
export const normal: React.SFC<{}> = () => (<LoadingOverlay />);
