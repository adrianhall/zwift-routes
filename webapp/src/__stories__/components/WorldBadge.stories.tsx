import React from 'react';
import WorldBadge from 'src/ui/components/WorldBadge';

export default {
  title: 'Components/WorldBadge'
};

// Sub-story - Bologna
export const bologna: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="Bologna" />
  </div>
);

// Sub-story - Crit City
export const critCity: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="Crit City" />
  </div>
);

// Sub-story - London
export const london: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="London" />
  </div>
);

// Sub-story - New York
export const newYork: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="New York" />
  </div>
);

// Sub-story - Richmond
export const richmond: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="Richmond" />
  </div>
);

// Sub-story - Watopia
export const watopia: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="Watopia" />
  </div>
);

// Sub-story - Yorkshire
export const yorkshire: React.SFC<{}> = () => (
  <div style={{ margin: '5rem;' }}>
    <WorldBadge world="Yorkshire" />
  </div>
);
