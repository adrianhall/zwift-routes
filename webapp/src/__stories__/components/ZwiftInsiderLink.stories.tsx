import React from 'react';
import ZwiftInsiderLink from 'src/ui/components/ZwiftInsiderLink';
import routeData from 'src/data/routes.json';
import { ZwiftRoute } from 'src/models';

export default {
  title: 'Components/ZwiftInsiderLink'
};

const route = new ZwiftRoute(routeData[0]);
const noRoute = new ZwiftRoute({ ...routeData[0], link: undefined });

// Sub-story - is not completed
export const hasLink: React.SFC<{}> = () => (
  <div style={{ margin: '5rem', padding: '4px', border: '1px solid black' }}>
    <ZwiftInsiderLink route={route} />
  </div>
);

// Sub-story - is not completed
export const noLink: React.SFC<{}> = () => (
  <div style={{ margin: '5rem', padding: '4px', border: '1px solid black' }}>
    <ZwiftInsiderLink route={noRoute} />
  </div>
);