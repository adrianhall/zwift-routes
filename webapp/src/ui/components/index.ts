export { default as CompletedIcon       } from './CompletedIcon';
export { default as LoadingIndicator    } from './LoadingIndicator';
export { default as LoadingOverlay      } from './LoadingOverlay';
export { default as ZwiftRouteList      } from './ZwiftRouteList';
export { default as ZwiftRouteListItem  } from './ZwiftRouteListItem';
export { default as ZwiftRouteTable     } from './ZwiftRouteTable';
