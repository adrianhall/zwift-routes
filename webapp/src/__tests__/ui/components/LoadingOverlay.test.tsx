import React from 'react';
import { render } from '@testing-library/react';
import { LoadingOverlay } from '../../../ui/components';

describe('LoadingOverlay', () => {
  // Snapshot test
  it('should render like the snapshot', () => {
    const { container } = render(<LoadingOverlay />);
    expect(container).toMatchSnapshot();
  });
});
