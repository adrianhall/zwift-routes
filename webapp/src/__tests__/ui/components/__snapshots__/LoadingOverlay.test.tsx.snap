// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay should render like the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="circle__Wrapper-sc-16bbsoy-0 fjQiTc"
      color="#fff"
      size="64"
    />
  </div>
</div>
`;
